pkg/controller/azure/virtualmachine/actuator.go

```diff
@@ -124,9 +142,24 @@ func (a *actuator) CreateOrUpdate(ctx context.Context, obj runtime.Object) (requ
 					RequeueAfter: a.config.RequeueInterval.Duration * (1 << (failedOperation.Attempts - 1)),
 				}
 			}
+
+			// If the configured max attempts has been reached, set VM states gauge to "failed" and return success
+			a.vmStatesGaugeVec.WithLabelValues(vmName).Set(VMStateFailed)
 			return 0, false, nil
 		}
 		azurev1alpha1.DeleteFailedOperation(&failedOperations, azurev1alpha1.OperationTypeReapplyVirtualMachine)
+
+		// Increase the reapplied VMs counter
+		a.reappliedVMsCounter.Inc()
+
+		// Set VM states gauge to "failed" or "ok" depending on the new Azure virtual machine state
+		a.setVMStatesGauge(reappliedAzureVM, vmName)
+	} else if azureVM != nil && getProvisioningState(azureVM) != compute.ProvisioningStateFailed {
+		// Set VM states gauge to "ok"
+		a.vmStatesGaugeVec.WithLabelValues(vmName).Set(VMStateOK)
+	} else if azureVM == nil {
+		// Delete VM states gauge
+		a.vmStatesGaugeVec.DeleteLabelValues(vmName)
```

Comment:
Yes, having in mind that MCM generates unique names every time I also think this is very unlikely. I think it's a good idea to delete gauges for VMs that are no longer present, otherwise we'll keep adding gauges and grafana graphs could start looking quite confusing.